### PR TITLE
Update webhook response handling to fix incoming message creation issue

### PR DESCRIPTION
**Pull request by vineyrawat**
### Description:
This pull request updates the webhook response handling to fix an issue where incoming messages were not being created. The commit message for this change is "Update webhook response handling to fix message creation issue", and it was made on the fix-incoming-message branch.

### Changes Made:

Modified the code to access incoming message data properly
Tested the changes to ensure that incoming messages are now being created correctly
Please review and merge this pull request if the changes look good. Thanks!

### CODE REVIEW

File: frappe_whatsapp/utils/webhook.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def post():
                     "message": message['text']['body']
                 }).insert(ignore_permissions=True)
     else:
-        update_status(data["entry"][0]["changes"][0])
+        update_status(data["entry"]["changes"][0])
```
Comment (shridarpatil) [inline frappe_whatsapp/utils/webhook.py:49]: `{"object": "whatsapp_business_account", "entry": [{"id": "107455662167439", "changes": [{"value": {"messaging_product": "whatsapp", "metadata": {"display_phone_number": "15550483859", "phone_number_id": "109288118647518"}, "contacts": [{"profile": {"name": "Shridhar"}, "wa_id": "xxxxx"}], "messages": [{"context": {"from": "15550483859", "id": "wamid.HBgMOTE5NzQxMDk0NDY4FQIAERgSMzUxQjUwOUY3OTREODJCNjE1AA=="}, "from": "xxxxxx", "id": "wamid.HBgMOTE5NzQxMDk0NDY4FQIAEhgWM0VCMDQ5OEEzRDYxOTRGMjYyODJCMgA=", "timestamp": "1680094862", "type": "interactive", "interactive": {"type": "button_reply", "button_reply": {"id": "2", "title": "Second Button\u2019s Titl"}}}]}, "field": "messages"}]}], "cmd": "frappe_whatsapp.utils.webhook.webhook"}`

This is the data which we get in webhook entry[0] is correct. The change you have made will not capture the message

Comment (vineyrawat) [reply frappe_whatsapp/utils/webhook.py:49]: In my case, I'm getting following response:
`{'object': 'whatsapp_business_account', 'entry': {'id': '103764272699039', 'changes': [{'value': {'messaging_product': 'whatsapp', 'metadata': {'display_phone_number': '15550256741', 'phone_number_id': '100188813063701'}, 'contacts': [{'profile': {'name': 'Vinay Rawat'}, 'wa_id': '919675705065'}], 'messages': [{'from': '919675705065', 'id': 'wamid.HBgMOTE5Njc1NzA1MDY1FQIAEhggNTIwQzBDQUJGMEI3NDAyNzcxQUEzMjZENzVBRkM3QzYA', 'timestamp': '1681798464', 'text': {'body': 'Raise an issue'}, 'type': 'text'}]}, 'field': 'messages'}]}, 'cmd': 'frappe_whatsapp.utils.webhook.webhook'}
`

Comment (shridarpatil) [reply frappe_whatsapp/utils/webhook.py:49]: which version your using?

Comment (vineyrawat) [reply frappe_whatsapp/utils/webhook.py:49]: Installed Apps
Call Log: v0.0.1 (vineet)
Core Erp: v0.0.1 (master)
ERPNext: v13.24.0 (version-13)
Frappe Framework: v13.24.0 (version-13)
Frappe Whatsapp: v0.0.2 (master)
Project: v0.0.1 (master)

Comment (vineyrawat) [reply frappe_whatsapp/utils/webhook.py:49]: Whatsapp Webhook v16

Comment (shridarpatil) [reply frappe_whatsapp/utils/webhook.py:49]: we need this to handle for both version of meta v15 and v16. Can you update the code to handle both cases

Comment (vineyrawat) [reply frappe_whatsapp/utils/webhook.py:49]: Sure, will update you soon : )

Comment (vineyrawat) [reply frappe_whatsapp/utils/webhook.py]: @shridarpatil Updated code for handling both v15 and v16

File: frappe_whatsapp/utils/webhook.py [hunk 2]
```diff
@@ -46,7 +51,12 @@ def post():
                     "message": message['text']['body']
                 }).insert(ignore_permissions=True)
     else:
-        update_status(data["entry"][0]["changes"][0])
+        try:
+            changes = data["entry"][0]["changes"][0]
+            update_status(changes)
+        except KeyError:
+            changes = data["entry"]["changes"][0]
+            update_status(changes)
```
Comment (shridarpatil) [inline frappe_whatsapp/utils/webhook.py:59]: add update_status outside try block so that we dont have redundant code

**Pull request merged**